compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/NodeParser.java

```diff
@@ -147,4 +147,15 @@ public static ModulePartNode parseModulePart(String text) {
         BallerinaParser parser = ParserFactory.getParser(text);
         return parser.parse().createUnlinkedFacade();
     }
+
+    /**
+     * Parses the input as a module part.
+     *
+     * @param text the input
+     * @return a {@code ModulePartNode}
+     */
+    public static Node parseIntermediateClause(String text) {
```

Comment:
We can narrow the return type to `IntermediateClauseNode` right?

Reply:
In the current code
if there is a error or null value
Following code will execute.
```

if (intermediateClause == null) {
            intermediateClause = SyntaxErrors.createMissingToken(SyntaxKind.SELECT_CLAUSE);
}
if (intermediateClause.kind == SyntaxKind.SELECT_CLAUSE) {
    STNode temp = intermediateClause;
    intermediateClause = SyntaxErrors.createMissingToken(SyntaxKind.SELECT_CLAUSE);
    intermediateClause = SyntaxErrors.cloneWithTrailingInvalidNodeMinutiae(intermediateClause, temp);
}
```

In that case these error nodes cannot convert to the `IntermediateClauseNode` Iguess

Reply:
ah no. the logic seems to be not correct. First of all, you can't return a select clause as an intermediate clause when the intermediateClause == null as according to the spec and our ST abstractions select clause does not fall under the intermediate clasues.

Reply:
createMissingToken() will only create a token. If we consider `where-clause` as suggested above, it is not a token with kind `WHERE_CLAUSE`. It is a node that requires two arguments. `createWhereClauseNode(whereKeyword, expression)`. 
Please refer to the `createMissingSimpleVarDecl(false)` on how to build the missing node.

Reply:
Updated.